workspace/api/openfdd_bridge/data_loader.py

```
@@ -47,6 +50,46 @@ def load_site_frame(
     return FeatherStore().read_site(site_id, source=source, columns=columns)
 
 
+def _merge_timeseries_frames(left: pd.DataFrame, right: pd.DataFrame) -> pd.DataFrame:
+    """Align irregular driver polls by nearest timestamp (30 min tolerance)."""
+    key_cols = {"timestamp", "site_id"}
+    extra = [c for c in right.columns if c not in left.columns and c not in key_cols]
+    if not extra:
+        return left
+    l = left.sort_values("timestamp").copy()
```

Ruff (0.15.15)

[error] Ambiguous variable name: `l`

(E741)